packages/wrangler/src/__tests__/dev.test.ts

```diff
@@ -1489,6 +1510,111 @@ describe.sequential("wrangler dev", () => {
 			);
 		});
 
+		it("should error when starting a dev session for an assets-only Worker with unsupported configuration for assets-only", async () => {
+			writeWranglerConfig({
+				assets: {
+					directory: "assets",
+				},
+				send_metrics: true,
+				vars: {
+					MY_VAR: "my var",
+				},
+				services: [
+					{
+						binding: "BINDING_NAME",
+						service: "WORKER_NAME",
+					},
+				],
+				kv_namespaces: [
+					{
+						id: "123",
+						binding: "KV_BINDING_NAME",
+					},
+				],
+			});
+			fs.mkdirSync("assets");
+			await expect(runWrangler("dev")).rejects
+				.toThrowErrorMatchingInlineSnapshot(`
+				[Error: Assets-only Workers do not support the following configuration keys:
+
+				⋅ "send_metrics"
+				⋅ "vars"
+				⋅ "kv_namespaces"
+				⋅ "services"
+
+				Please remove these fields from your configuration file, or configure the "main" field if you are trying to deploy a Worker with assets.]
+			`);
+		});
+
+		it("should error when starting a dev session for an assets-only Worker with the --assets flag set but with unsupported configuration for assets-only", async () => {
+			writeWranglerConfig({
+				send_metrics: true,
+				vars: {
+					MY_VAR: "my var",
+				},
+				services: [
+					{
+						binding: "BINDING_NAME",
+						service: "WORKER_NAME",
+					},
+				],
+				kv_namespaces: [
+					{
+						id: "123",
+						binding: "KV_BINDING_NAME",
+					},
+				],
+			});
+			fs.mkdirSync("assets");
+			await expect(runWrangler("dev --assets assets")).rejects
+				.toThrowErrorMatchingInlineSnapshot(`
+				[Error: Assets-only Workers do not support the following configuration keys:
+
+				⋅ "send_metrics"
+				⋅ "vars"
+				⋅ "kv_namespaces"
+				⋅ "services"
+
+				Please remove these fields from your configuration file, or configure the "main" field if you are trying to deploy a Worker with assets.]
+			`);
+		});
+
+		it.fails(
```

Comment:
related to https://github.com/cloudflare/workers-sdk/pull/8998/files#r2048784806